fix(Dropdown): infinite render onChange, positioning in flex

saurabhdaware opened the pull request:
Fixes 2 bugs
- [Inifinite render](https://razorpay.slack.com/archives/CMQ3RBHEU/p1676970701130639?thread_ts=1676869625.828389&cid=CMQ3RBHEU)
- [positioning in display flex container](https://razorpay.slack.com/archives/CMQ3RBHEU/p1676871892270009?thread_ts=1676869625.828389&cid=CMQ3RBHEU)

saurabhdaware commented on packages/blade/src/components/Dropdown/Dropdown.stories.tsx:
```diff
@@ -510,13 +510,15 @@ export const WithValueDisplay = (args: AllDropdownProps): JSX.Element => {
           label="Select Action"
-          onChange={({ values }) => {
-            setDropdownValues(values.join(', '));
+          onChange={({ name, values }) => {
```
it was working when state was string but it was breaking when state was object. Changed it in the story as well so we catch these issues.

saurabhdaware commented on packages/blade/src/components/Input/SelectInput/SelectInput.tsx:
```diff
@@ -66,7 +66,8 @@ const _SelectInput = (
   React.useEffect(() => {
     onChange?.({ name: props.name, values: value.split(', ') });
-  }, [value, onChange, props.name]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value, props.name]);
```
Having `onChange` in this array was making the function trigger on parent re-renders which was causing infinite re-renders.

sudhakar-selva replied:
[useEvent](https://github.com/reactjs/rfcs/blob/useevent/text/0000-useevent.md) hook might be useful here

saurabhdaware replied:
yup

divyanshu013 commented on packages/blade/src/components/Dropdown/Dropdown.stories.tsx:
```diff
@@ -720,4 +722,52 @@ WithRefUsage.args = {
+          </DropdownOverlay>
+        </Dropdown>
+      </BaseBox>
+      <BaseBox flex={1}>
```
Minor: can perhaps add some spacing in between

saurabhdaware replied:
done

divyanshu013 reviewed (approved):
Looks good to me.

saurabhdaware commented:
@kamleshchandnani can we merge this?

kamleshchandnani reviewed (commented):
overall LGTM, one thing I would recommend is that
1. Can we explicitly add stories for controlled vs uncontrolled?
2. What are the remaining props of dropdown/select that we aren't part of the storybook we shall add all of them.

saurabhdaware commented:
> Can we explicitly add stories for controlled vs uncontrolled?

We don't exactly have fully uncontrolled dropdown right now. For that we'll need `useDropdown` hook that can give functions like `selectOption` (this can be used for that keyboard bindings usecase we had discussed). 

We do have `isDefaultSelected` prop that sets default selection (I have added it in #1020. It was missing)

> What are the remaining props of dropdown/select that we aren't part of the storybook we shall add all of them.

Added it here #1020 (this is a bit of temporary solution though. not fully satisfied with docs right now. I'll fix it all when I am adding Button trigger and other things to docs)
